scripts/fhap_build.R

```diff
@@ -1,24 +1,77 @@
 # build the FHAP digital form
 
 source('scripts/packages.R')
-source('scripts/functions.R')
 
-form_raw <- readxl::read_xlsx('data/FHAP_template.xlsx')
+#' name the project directory we are burning to
+dir_project <- 'bcfishpass_20230516'
 
-form_types <- readxl::read_xlsx('data/FHAP_template.xlsx',
-                                skip = 4)
+#' name the form using the date and time
+#' we should be able to name the form the same in the active project but the files can be versioned
+file_name <- paste0('form_fhap_', format(lubridate::now(), "%Y%m%d"))
 
 
-cols_to_add <- c('stream',
-                 'date',
-                 'utm_start',
-                 'utm_end',
-                 'surveyor_1',
-                 'surveyor_2')
+#' define your utm zone.  This can cause errors if you use the form in more than
+#' one zone!!!!! beware
+utm_zone <- 11
 
-form_col_names <- form_raw %>%
-  wkb_col_names() %>%
-  stringr::str_replace_all(., 'd_s_utm_', '')
+#' import the fhap submission template (needs to be in the data directory)
+#' because we want to keep the backup file clean for the value maps and because
+#' we are not worried about version controlling this data we turn the `backup` function to `FALSE`
 
-# this will make a tibble but maybe we make one with populated options and let it guess
-form_col_names %>% purrr::map_dfc(setNames, object = list(character()))
+form_raw <- readxl::read_xlsx('data/FHAP_template.xlsx', skip = 1)
+
+# see the names of the columns
+names(form_raw)
+
+fhap <- form_raw %>%
+  janitor::clean_names() %>%
+  # add some columns of our own
+  dplyr::mutate(stream = NA_character_,
+                weather = NA_character_,
+                date = NA_POSIXct_,
+                waypoint_start = NA_integer_,
+                waypoint_end = NA_integer_,
+                crew_members = NA_character_,
+                temperature = as.numeric(NA_character_),
+                conductivity = as.numeric(NA_character_),
+                pH = as.numeric(NA_character_),
+                disturbance2 = NA_character_,
+                disturbance3 = NA_character_,
+                photo_top = NA_character_,
+                photo_bottom = NA_character_,
+                photo_extra_1 = NA_character_,
+                photo_extra_1_tag = NA_character_,
+                photo_extra_2 = NA_character_,
+                photo_extra_2_tag = NA_character_,
+                method_site_card = NA_character_
+  ) %>%
+  # column QA
+  # there are columns with similar names because they are used in different groups, specify group in column name, then group columns in mergin form
+  rename(type_hab_unit = type_5,
```

Comment:
there are ways to grab column names that will append the merged row columns with the individual column names (ex. location_site, location_reach_number, etc.).  Have a look at the blackwater scripts to see how it is done

we want to build scripts that allow us flexibility for when things in our template change.  Directly renaming things should be a very last resort

Reply:
Thank you I was wondering about that. Tried a few things out but couldn't get it to work. I'll take a look there.